src/hooks/useSearch.tsx

```diff
@@ -0,0 +1,10 @@
+import { useMemo } from "react";
+import {ISearchingResult} from "../types/types";
+
+export const useSearch = (query: string, list: ISearchingResult[]): ISearchingResult[] => {
+    const searchingResults = useMemo(() => {
```

Comment:
Missed this last time. Here better use useCallback hook and use `query` as an input parameter. Because in our case we have static (and in theory large) list and changeable query.
For the next task with api both parameters would be changeable and list would barely have more then 20. In this case no optimization needed. Also incoming values would always be filtered. Don't forget to take this into account.

Reply:
You don't need to fix it now as it should be changed in the next task